webpack/scenes/ContentViews/Details/ContentViewDetailActions.js

```diff
@@ -51,6 +57,37 @@ export const updateContentView = (cvId, params) => async dispatch => dispatch(pu
   },
 }));
 
+export const addComponent = params => async dispatch => dispatch(put({
+  type: API_OPERATIONS.PUT,
+  key: cvDetailsKey(params.compositeContentViewId),
+  url: api.getApiUrl(`/content_views/${params.compositeContentViewId}/content_view_components/${params.components.id ? params.components.id : 'add'}`),
+  params: params.components.id ? params.components : params,
+  handleSuccess: () => cvUpdateSuccess({
+    data: { id: params.compositeContentViewId },
+  }, dispatch),
+  handleError: error => dispatch(apiError(null, error)),
+  actionTypes: {
+    REQUEST: ADD_COMPONENT,
+    SUCCESS: ADD_COMPONENT_SUCCESS,
+    FAILURE: ADD_COMPONENT_FAILURE,
+  },
```

Comment:
Specifying action types isn't necessary unless you want to customize the names.  In this case it actually withholds information:

If I comment this out I get action types of `CONTENT_VIEWS_COMPONENTS_5_REQUEST` and `CONTENT_VIEWS_COMPONENTS_5_SUCCESS`.

But with the custom action types it's just `ADD_COMPONENT` and `ADD_COMPONENT_SUCCESS`

Unless we really want to show a different name in the Redux log, I think we should remove the `actionTypes` everywhere.  We should be more clear about that in the usage docs..

Reply:
Removed..Using a special key and removing these..Let me know how it looks now.